### PR TITLE
PAYARA-3356 EJB Timer fails when using non-persistent flag with @Timeout (Payara 4.x)

### PR DESCRIPTION
**Pull request by jGauravGupta**

### CODE REVIEW

**Comment by jGauravGupta** [comment]
jenkins test please

**Comment by jGauravGupta** [comment]
jenkins test please

File: appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/EJBTimerService.java [hunk 1]
```diff
@@ -200,7 +200,7 @@ public static EJBTimerServiceWrapper getEJBTimerServiceWrapper(EJBContextImpl ej
     }
 
     protected static void setPersistentTimerService(EJBTimerService timerService) {
-        persistentTimerService = timerService;
+         persistentTimerService = timerService;
```
Comment (Pandrex247) [inline appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/EJBTimerService.java]: (╯°□°）╯︵ ┻━┻

File: appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/BaseContainer.java [hunk 2]
```diff
@@ -824,8 +828,10 @@ protected BaseContainer(ContainerType type, EjbDescriptor ejbDesc, ClassLoader l
                     if (_logger.isLoggable(Level.FINE)) {
                         _logger.log(Level.FINE, "... processing {0}", method);
                     }
-                    if (!isPersistenceTimer) {
-                        isPersistenceTimer = schd.getPersistent();
+                    if(schd.getPersistent()) {
```
Comment (arjantijms) [inline appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/BaseContainer.java]: (╯°□°）╯︵ ┻━┻

**Pull request merged**